### ISSUE TITLE
ARM64 node of builder not used

### ISSUE THREAD

**Issue by skaldesh** [docker/buildx#1376]
Hi, I have setup a new builder instance that includes two nodes, one on AMD64 (local) and the other a native ARM64 node.  
This is the output of `docker buildx ls`:  
```
builder1-cluster docker-container
  builder1       local            running v0.10.5  linux/amd64, linux/amd64/v2, linux/amd64/v3, linux/amd64/v4, linux/386
  builder-arm1   builder-arm1     running v0.10.5  linux/arm64, linux/arm/v7, linux/arm/v6
builder-arm1     docker
  builder-arm1   builder-arm1     running 20.10.7  linux/arm64, linux/arm/v7, linux/arm/v6
default *        docker
  default        default          running 20.10.20 linux/amd64, linux/386
local            docker
  local          local            running 20.10.20 linux/amd64, linux/386
```

However, when I run `docker buildx --builder builder1-cluster bake --progress plain --push builder-base` I notice that only the AMD64 machine is actually doing something (I observed both machines with `htop`, nothing is happening on the ARM64 node).  

In addition, the build for `linux/arm64` always fails with these messages:  
```
#13 103.4 debconf: delaying package configuration, since apt-utils is not installed
#13 103.5 Fetched 241 MB in 9s (26.9 MB/s)
#13 103.5 Error while loading /usr/sbin/dpkg-split: No such file or directory
#13 103.5 Error while loading /usr/sbin/dpkg-deb: No such file or directory
#13 103.5 dpkg: error processing archive /var/cache/apt/archives/libpython3.10-minimal_3.10.6-1~22.04_arm64.deb (--unpack):
#13 103.5  dpkg-deb --control subprocess returned error exit status 1
#13 103.5 Error while loading /usr/sbin/dpkg-split: No such file or directory
#13 103.5 Error while loading /usr/sbin/dpkg-deb: No such file or directory
#13 103.5 dpkg: error processing archive /var/cache/apt/archives/libexpat1_2.4.7-1_arm64.deb (--unpack):
#13 103.5  dpkg-deb --control subprocess returned error exit status 1
#13 103.5 Error while loading /usr/sbin/dpkg-split: No such file or directory
#13 103.5 Error while loading /usr/sbin/dpkg-deb: No such file or directory
#13 103.5 dpkg: error processing archive /var/cache/apt/archives/python3.10-minimal_3.10.6-1~22.04_arm64.deb (--unpack):
#13 103.5  dpkg-deb --control subprocess returned error exit status 1
#13 103.6 Errors were encountered while processing:
#13 103.6  /var/cache/apt/archives/libpython3.10-minimal_3.10.6-1~22.04_arm64.deb
#13 103.6  /var/cache/apt/archives/libexpat1_2.4.7-1_arm64.deb
#13 103.6  /var/cache/apt/archives/python3.10-minimal_3.10.6-1~22.04_arm64.deb
#13 104.0 E: Sub-process /usr/bin/dpkg returned an error code (1)

...

#24 [linux/arm64 final  6/31] RUN <<EOF (mkdir -p /usr/local/share/ca-certificates/extra...)
#24 0.154 /bin/sh: 3: update-ca-certificates: not found
#24 ERROR: process "/dev/.buildkit_qemu_emulator /bin/sh -c mkdir -p /usr/local/share/ca-certificates/extra\nmv /CA.pem /usr/local/share/ca-certificates/extra/CA.crt\nupdate-ca-certificates\n" did not complete successfully: exit code: 127
```

Notice the `ERROR: process "/dev/.buildkit_qemu_emulator"`! Why is it using QEMU? I have not set that up.  

Bake file:
```
variable "TAG" {
    default = "dev"
}

### Builder
target "builder-base" {
    context = "."
    dockerfile = "builder/.base.dockerfile.gen"
    platforms = ["linux/amd64", "linux/arm64"]
    tags = ["docker.company.m/nlab/builder:${TAG}"]
    pull = true
}
```

This is the output of `docker version`:  
AMD64
```
Client:
 Version:           20.10.20
 API version:       1.41
 Go version:        go1.19.2
 Git commit:        9fdeb9c3de
 Built:             Sat Oct 22 19:31:23 2022
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server:
 Engine:
  Version:          20.10.20
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.19.2
  Git commit:       03df974ae9
  Built:            Sat Oct 22 19:30:29 2022
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.6.8
  GitCommit:        9cd3357b7fd7218e4aec3eae239db1f68a5a6ec6.m
 runc:
  Version:          1.1.4
  GitCommit:
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

ARM64
```
Client:
 Version:           20.10.7
 API version:       1.41
 Go version:        go1.13.8
 Git commit:        20.10.7-0ubuntu5~18.04.3
 Built:             Mon Nov  1 01:04:31 2021
 OS/Arch:           linux/arm64
 Context:           default
 Experimental:      true

Server:
 Engine:
  Version:          20.10.7
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.8
  Git commit:       20.10.7-0ubuntu5~18.04.3
  Built:            Fri Oct 22 00:57:37 2021
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.5.5-0ubuntu3~18.04.2
  GitCommit:
 runc:
  Version:          1.0.1-0ubuntu2~18.04.1
  GitCommit:
 docker-init:
  Version:          0.19.0
  GitCommit:
```

**Comment by wojiushixiaobai** [comment]
try
```bash
docker buildx use builder1-cluster
```
or
```bash
docker buildx use builder-arm1
```

**Comment by skaldesh** [comment]
I found the solution, it was similar to my last issue's setup: https://github.com/docker/buildx/issues/1084  

I started the build in a docker container and this time, I mounted the wrong ssh-agent socket into the container. The container was therefore not able to connect via SSH to the ARM builder and just started building locally instead.

IMHO, I think some error message would be appropriate, since the local builder instance reports in `docker buildx ls` that is does not support the `linux/arm64` platform. If I want to bake a target with such a platform and there is no builder available for it (in my case, it would not be accessed), the build should FAIL.  

Otherwise, the error is really hard to spot.
